pilot/pkg/bootstrap/server.go

```diff
@@ -177,6 +178,18 @@ type Server struct {
 	mux *http.ServeMux
 }
 
+func createInterface(kubeconfig string) (kubernetes.Interface, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.ExplicitPath = kubeconfig
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(restConfig)
+}
+
```

Comment:
We probably ought to use this pattern in [`pilot/pkg/serviceregistry/kube/client.go:CreateInterface`](https://github.com/istio/istio/blob/f5208b523a9471bb49a6f9a0f2c73a0bcc9c9b2b/pilot/pkg/serviceregistry/kube/client.go#L73) and [`pilot/pkg/config/kube/crd/client.go:CreateRESTConfig`](https://github.com/istio/istio/blob/f157c74ebfcf9693f5357787925ac91a3a0caf03/pilot/pkg/config/kube/crd/client.go#L169) as well.